### PR TITLE
LTP: Update ltp-open-posix to run specific group test

### PR DESCRIPTION
**Pull request by Lakshmipathi**
Run specific group test from ltp-open-posix. Parse the
results and send it to lava-test-case.

Signed-off-by: lakshmipathi ganapathi <lakshmipathi.ganapathi@collabora.com>

### CODE REVIEW

**Comment by gctucker** [comment]
Here's some sample results from the open POSIX timers tests using this patch:
https://linux.kernelci.org/test/plan/id/60e2ae83e8c2492bdc11796e/ (x86 AMD)
https://linux.kernelci.org/test/plan/id/60e2b414ad077537cc11796a/ (arm)

**Review by nareshkamboju — APPROVED** [review]

File: automated/linux/ltp-open-posix/ltp-open-posix.sh [hunk 1]
```diff
@@ -63,6 +66,16 @@ parse_ltp_output() {
 
 }
 
+# Parse LTP open posix grp output
+parse_ltp_grp_output() {
+    sed -i '/^PASS\|^FAIL\|^Testing/!d' ${GRP_OUTPUT_FILE}
```
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.sh:71]: LTP open posix test suites results verdicts / Labels are different than pass, fail and skip.
Please consider below labels also.
SKIPPED|: UNSUPPORTED|: UNTESTED|: UNRESOLVED|: HUNG

Comment (Lakshmipathi) [reply automated/linux/ltp-open-posix/ltp-open-posix.sh:71]: Okay. Let me add these labels too.

**Comment by nareshkamboju** [comment]
@Lakshmipathi Please fix shellcheck warnings / errors.

**Comment by Lakshmipathi** [comment]
@nareshkamboju , sure just now update the patch to resolve spellcheck issue. Will look into labels.

File: automated/linux/ltp-open-posix/ltp-open-posix.sh (hunk 1 above)
Comment (Lakshmipathi) [reply automated/linux/ltp-open-posix/ltp-open-posix.sh]: Handled these labels and tested the changes with this job: https://lava.collabora.co.uk/scheduler/job/4156348

**Comment by gctucker** [comment]
@nareshkamboju It looks like you or some maintainer needs to give approval for the GitHub checks to run, for first-time contributions.

**Comment by Lakshmipathi** [comment]
Updated the patch to resolve recent linting errors

File: automated/linux/ltp-open-posix/ltp-open-posix.yaml [hunk 2]
```diff
@@ -22,9 +22,8 @@ params:
     # LTP version
     LTP_VERSION: 20180515
     SKIP_INSTALL: true
-
 run:
     steps:
         - cd ./automated/linux/ltp-open-posix/
-        - ./ltp-open-posix.sh -s "${SKIP_INSTALL}" -v "${LTP_VERSION}"
+        - ./ltp-open-posix.sh -s "${SKIP_INSTALL}" -v "${LTP_VERSION}" -t "${GRP_TEST}"
```
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.yaml:28]: @Lakshmipathi Please describe the purpose of "GRP_TEST" in params: section.

**Comment by Lakshmipathi** [comment]
@nareshkamboju could you please trigger linting check? This time it should pass.

**Comment by bhcopeland** [comment]
@Lakshmipathi triggered

**Comment by Lakshmipathi** [comment]
thanks @bhcopeland

**Review by nareshkamboju — APPROVED** [review]

File: automated/linux/ltp-open-posix/ltp-open-posix.yaml [hunk 3]
```diff
@@ -22,9 +22,10 @@ params:
     # LTP version
     LTP_VERSION: 20180515
     SKIP_INSTALL: true
-
+    # Run Option Group Feature Tests. Available Groups are AIO, MEM, MSG, SEM, SIG, THR, TMR, TPS
+    GRP_TEST: ""
```
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.yaml:26]: Add an extra empty line

File: automated/linux/ltp-open-posix/ltp-open-posix.sh [hunk 4]
```diff
@@ -72,8 +92,18 @@ run_ltp_open_posix() {
       make "${EACH_TEST}"-test || true
     done
     parse_ltp_output
+
```
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.sh:95]: Is this empty line needed here ?

File: automated/linux/ltp-open-posix/ltp-open-posix.sh [hunk 5]
```diff
@@ -72,8 +92,18 @@ run_ltp_open_posix() {
...
 
+# Run specific LTP open posix group tests
+run_ltp_open_posix_grp() {
+    # shellcheck disable=SC2164
+    GRP_TEST=$1
+    "${LTP_PRE_INSTALL}"/bin/run-posix-option-group-test.sh "${GRP_TEST}" | tee "${GRP_OUTPUT_FILE}"
+    parse_ltp_grp_output
+}
+
+
```
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.sh:106]: you may delete this empty line.

File: automated/linux/ltp-open-posix/ltp-open-posix.sh [hunk 6]
```diff
@@ -63,6 +66,23 @@ parse_ltp_output() {
...
+    sed -i 's/\s\s*/ /g' "${GRP_RESULT_FILE}"
+    sed -i 's/Testing //g' "${GRP_RESULT_FILE}"
+    {
+    grep -E "FAIL 0" "${GRP_RESULT_FILE}" | awk '{print $1 " " $2}' | sed 's/PASS/pass/; s/FAIL/fail/; s/CONF/skip/';
+    grep -v "FAIL 0" "${GRP_RESULT_FILE}" | awk '{print $1 " " $4}' | sed 's/PASS/pass/; s/FAIL/fail/; s/CONF/skip/';
+    grep -e "UNRESOLVED" -e "UNTESTED" -e "UNSUPPORTED" -e "HUNG" -e "SKIPPED" "${GRP_OUTPUT_FILE}.bak" \
+	    | awk -F':' '{print $1 $3}' \
+	    | awk -F'/' '{print $NF}' \
+	    | sed 's/SKIPPED/skip/; s/UNSUPPORTED/fail/; s/UNTESTED/fail/; s/UNRESOLVED/fail/; s/HUNG/skip/'
+    } >> "${RESULT_FILE}"
```
Comment (nareshkamboju) [inline automated/linux/ltp-open-posix/ltp-open-posix.sh:82]: Please share the latest LAVA test log links with this later code.

**Review by nareshkamboju — CHANGES_REQUESTED** [review]
Please check inline comments.

File: automated/linux/ltp-open-posix/ltp-open-posix.sh (hunk 6 above)
Comment (Lakshmipathi) [reply automated/linux/ltp-open-posix/ltp-open-posix.sh:82]: ok, here is the LAVA job with these changes: https://lava.collabora.co.uk/scheduler/job/4189498

**Comment by Lakshmipathi** [comment]
Updated PR and resolved latest review comments.

**Comment by nareshkamboju** [comment]
+1
Looks good to me.

**Review by nareshkamboju — APPROVED** [review]

File: automated/linux/ltp-open-posix/ltp-open-posix.sh [hunk 7]
```diff
@@ -63,6 +66,22 @@ parse_ltp_output() {
...
 
+# Parse LTP open posix grp output
+parse_ltp_grp_output() {
+    sed -i.bak '/^PASS\|^FAIL\|^Testing/!d' "${GRP_OUTPUT_FILE}"
+    paste -d " "  - - - < "${GRP_OUTPUT_FILE}" | tee "${GRP_RESULT_FILE}"
+    sed -i 's/\s\s*/ /g' "${GRP_RESULT_FILE}"
+    sed -i 's/Testing //g' "${GRP_RESULT_FILE}"
+    {
+    grep -E "FAIL 0" "${GRP_RESULT_FILE}" | awk '{print $1 " " $2}' | sed 's/PASS/pass/; s/FAIL/fail/; s/CONF/skip/';
+    grep -v "FAIL 0" "${GRP_RESULT_FILE}" | awk '{print $1 " " $4}' | sed 's/PASS/pass/; s/FAIL/fail/; s/CONF/skip/';
```
Comment (roxell) [inline automated/linux/ltp-open-posix/ltp-open-posix.sh:77]: Can we squash these two lines into one?
something like this:
| sed 's/PASS/pass/; s/FAIL/fail/; s/CONF/skip/'|awk '{ if($5 == 1) {print $1 " " $4} else {print $1 " " $2} }'

Comment (Lakshmipathi) [reply automated/linux/ltp-open-posix/ltp-open-posix.sh]: @roxell thanks. Updated the PR to use single line and tested the change. https://lava.collabora.co.uk/scheduler/job/4202810

**Comment by Lakshmipathi** [comment]
Could someone please trigger shellcheck for recent change?

**Pull request merged**